pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -449,6 +450,16 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[string]b
 	return localityLbEndpoints
 }
 
+// addUint32AvoidOverflow returns sum of two uint32.
+// If sum overflows, and the MaxUint32 returns.
+func addUint32AvoidOverflow(left, right uint32) uint32 {
+	newVal := left + right
+	if newVal < left || newVal < right {
+		return math.MaxUint32
```

Comment:
Not overflowing is definitely preferable, but since this would probably throw off some weights, is it possible to either:

* Normalize using some GCD algorithm
* Log a warning

Reply:
Yes, I have recorded the corresponding warning log.